python/pyarrow/serialization.pxi

```diff
@@ -119,6 +119,11 @@ cdef class SerializationContext:
             This argument is optional, but can be provided to
             deserialize objects of the class in a particular way.
         """
+        if not isinstance(type_id, six.string_types):
+            raise ValueError("The type_id argument must be a string. The "
```

Comment:
This should raise `TypeError` instead

Reply:
Either should be ok

Reply:
I don't have a preference, I'll change it.

Reply:
Fixed.

Reply:
Thanks